extensions/kind/src/image-handler.ts

```diff
@@ -43,33 +50,55 @@ export class ImageHandler {
         { placeHolder: 'Select a Kind cluster to push to' },
       );
     }
+
+    // Only proceed if a cluster was selected
     if (selectedCluster) {
-      const filename = await tmpName();
+      let name = image.name;
+      let filename: string;
+      const env = process.env;
+
+      // Create a name:tag string for the image
+      if (image.tag) {
+        name = name + ':' + image.tag;
+      }
+
+      // Check to see if the selected cluster is either podman or docker
+      if (selectedCluster.engineType === 'podman') {
+        env['KIND_EXPERIMENTAL_PROVIDER'] = 'podman';
+      } else {
+        env['KIND_EXPERIMENTAL_PROVIDER'] = 'docker';
+      }
+
       try {
-        let name = image.name;
-        if (image.tag) {
-          name = name + ':' + image.tag;
-        }
+        // Create a temporary file to store the image
+        filename = await tmpName();
+
+        // Save the image to the temporary file
         await extensionApi.containerEngine.saveImage(image.engineId, name, filename);
-        const env = process.env;
-        if (selectedCluster.engineType === 'podman') {
-          env['KIND_EXPERIMENTAL_PROVIDER'] = 'podman';
-        } else {
-          env['KIND_EXPERIMENTAL_PROVIDER'] = 'docker';
-        }
+
+        // Run the Kind load command to push the image to the cluster
         await runCliCommand(kindCli, ['load', 'image-archive', '-n', selectedCluster.label, filename], {
           env: env,
         });
-        extensionApi.window.showNotification({
-          body: `Image ${image.name} pushed to Kind cluster ${selectedCluster.label}`,
-        });
-      } catch (error) {
-        const errorMessage = error instanceof Error ? error.message : '' + error;
-        throw new Error(
-          `Error while pushing image ${image.name} to Kind cluster ${selectedCluster.label}: ${errorMessage}`,
+
+        // Show a dialog to the user that the image was pushed
+        // TODO: Change this to taskbar notification when implemented
+        await extensionApi.window.showInformationMessage(
```

Comment:
Are all async tasks going to show a message now ? @mairin WDYT ?

Reply:
@jeffmaury Ideally it'd be a taskbar notice instead of a dialog (the comment indicates that too). But yeh I think all async tasks should be in the task notification area. Does that make sense?